mypy/plugins/default.py

```diff
@@ -313,14 +313,18 @@ def typed_dict_setdefault_signature_callback(ctx: MethodSigContext) -> CallableT
         isinstance(ctx.type, TypedDictType)
         and len(ctx.args) == 2
         and len(ctx.args[0]) == 1
-        and isinstance(ctx.args[0][0], StrExpr)
         and len(signature.arg_types) == 2
         and len(ctx.args[1]) == 1
     ):
-        key = ctx.args[0][0].value
-        value_type = ctx.type.items.get(key)
-        if value_type:
-            return signature.copy_modified(arg_types=[str_type, value_type])
+        key = None
+        if isinstance(ctx.args[0][0], StrExpr):
+            key = ctx.args[0][0].value
+        if isinstance(ctx.args[0][0], NameExpr):
+            key = ctx.args[0][0].name
```

Comment:
I think you should use `arg_types` from `ctx` and use `try_getting_str_literals_from_type` to get the literal values of the argument. There are some examples in this file that do a similar thing.